hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/WriteStatus.java

```diff
@@ -83,16 +86,20 @@ public WriteStatus() {
     this.random = null;
   }
 
+  public void markSuccess() {
+    totalRecords++;
```

Comment:
This is only used by `HoodieRowCreateHandle`, not sure whether we need to track the location changes.

Reply:
so this was done in that way mainly due to hoodie key is not applicable when using spark bulkinsert row-writing without metafields. but to consolidate APIs and track more info like new location, we can just use the other api with dummy key